docs/treefile.md

```diff
@@ -104,9 +104,18 @@ It supports the following parameters:
     `--no-hostonly` will be used.
 
  * `rpmdb`: String, optional: The RPM database backend.  Can be one of
-    `bdb`, `ndb`, or `sqlite`.  If unspecified, defaults to `bdb` for
-    compatibility. The default can be set to `sqlite` at compile-time
-    via `--enable-sqlite-rpmdb-default`.
+    `target`, `bdb`, `ndb`, or `sqlite`.  This option is a historical
+    mistake; ultimately the only thing that really works is to write
+    the rpmdb in the `target` format - the format that the `librpm`
+    library in the target filesystem tree understands.  Longer term,
+    that will become the default.
+    However, when not set, the current default is to try to match
+    the host environment's `librpm`.
```

Comment:
It looks like we are going out of our way to basically say this default is wrong, and it should be matching the target instead.
Is there something preventing us to cut a step here and directly fix this default instead of introducing another variant?

Reply:
It's a good question.  Basically it's conservatism - we're making a fundamental change opt-in to reduce risk.

We could instead try adding `rpmdb: host` and make `rpmdb: target` the default.   That would allow users to back out the change if there's a problem.

Concretely the case I got uncertain about is that we're still shipping newer rpm-ostree to Fedora 33, where the default is still bdb.  But I think that's *probably* fine because we'd still continue to use bdb.

Reply:
OK reworked to do that.